mobile-app/src/shared/hooks/use-location.ts

```diff
@@ -15,24 +17,99 @@ export interface DeviceLocation {
 }
 
 const useLocation = () => {
-	const [permissionGranted, setPermissionGranted] = useState(false);
+	const [permissionResponse, setPermissionResponse] = useState<Location.LocationPermissionResponse>();
+	const [location, setLocation] = useState({} as DeviceLocation);
+	const appState = useRef(AppState.currentState);
+	const locationWatcher = useRef<LocationSubscription | null>(null);
 
-	const requestPermissionGetLocationAsync = async (resolveLocation: () => Promise<Location.LocationObject | null>): Promise<DeviceLocation> => {
-		const { status } = await Location.requestForegroundPermissionsAsync();
+	useEffect(() => {
+		requestPermissionAndExecute(async () => {
+			await setLastKnownLocation();
+			startLocationTracking();
+		});
+		const appStateSubscription = AppState.addEventListener('change', handleAppStateChange);
 
-		if (status !== 'granted') {
-			setPermissionGranted(false);
-			return {} as DeviceLocation;
-		} 
+		return () => {
+			stopLocationTracking();
+			appStateSubscription.remove();
+		}
+	}, []);
 
-		setPermissionGranted(true);
-		return getLocation(resolveLocation);
+	const getWatcherOptions = async (): Promise<{}> => {
+		const batteryLevel = await getBatteryLevelAsync();
+		return {
+			accuracy: Location.Accuracy.Highest,
```

Comment:
Mozemo i smanjiti accuracy za stepeb ako je ispo 10%, jer i to utice na bateriju.